controllers/core/openstackcontrolplane_controller.go

```diff
@@ -133,13 +131,9 @@ func (r *OpenStackControlPlaneReconciler) Reconcile(ctx context.Context, req ctr
 	// Reset all ReadyConditons to 'Unknown'
 	instance.InitConditions()
 
-	// If we're not deleting this and the service object doesn't have our finalizer, add it.
-	if instance.DeletionTimestamp.IsZero() && controllerutil.AddFinalizer(instance, helper.GetFinalizer()) {
-		return ctrl.Result{}, err
-	}
-
 	if !instance.DeletionTimestamp.IsZero() {
-		return r.reconcileDelete(ctx, instance, helper)
+		r.Log.Info("Reconciled OpenStackControlPlane delete successfully")
+		return ctrl.Result{}, err
```

Comment:
Based on my tests we never reach this code path now. Probably because there is no finalizer on the OpenstackControlPlane resource, the delete of the resource is automatic and even if the reconciler is called it will get NotFound from the API why trying to read the instance. As we are not logging there the `make openstack_cleanup_deploy` target actually logs nothing in the openstack-operator.

Reply:
Proposed cleanup and log move in https://github.com/openstack-k8s-operators/openstack-operator/pull/229

Reply:
Hmm.. look like if custom resource does not have any finalizers, the controller will not be involved in the deletion/finalization process. I thought metdata deletionTimestamp would anyway be set and it's possible for controller reconcile to run before it's been garbage collected.